app/harvesters/loc_harvester.rb

```diff
@@ -0,0 +1,174 @@
+##
+# Harvester for Library of Congress
+##
+
+class LocHarvester < Krikri::Harvesters::ApiHarvester
+ include Krikri::Harvester
+  attr_reader :opts
+
+  ##
+  #
+  # @param opts [Hash] options for the harvester
+  # @see .expected_opts
+  #
+  # The options for this harvester should be an array of URIs because LoC is
+  # specifying sets of records to be harvested.
+  #
+  def initialize(opts = {})
+    # https://www.loc.gov/collections/american-revolutionary-war-maps/ (1,251 items)
+    # https://www.loc.gov/collections/civil-war-maps/ (1,707 items)
+    # https://www.loc.gov/collections/panoramic-maps/ (1,433 items)
+    opts[:uri] ||= ['https://www.loc.gov/collections/american-revolutionary-war-maps/?fo=json',
+      'https://www.loc.gov/collections/civil-war-maps/?fo=json',
+      'https://www.loc.gov/collections/panoramic-maps/?fo=json']
+    # TODO add name (check)
+    opts[:name] ||= 'loc'
+    # Item URI
+    opts[:item_uri] ||= 'https://www.loc.gov/item/'
+    super
+  end
+
+  ##
+  # @return [Hash] A hash documenting the allowable options to pass to
+  #   initializers.
+  #
+  # @see Krikri::Harvester::expected_opts
+  def self.expected_opts
+    {
+      key: :api,
+        opts: {
+       }
+    }
+  end
+
+  ##
+  # @param identifier [#to_s] the identifier of the record to get
+  # @return [#to_s] the record
+  def get_record(identifier)
+    response = request(opts[:item_uri] + identifier.to_s + '?fo=json')
+    build_record( response )
+  end
+
+
+  private
+
+  ##
+  # @param doc [Hash] the partial item content included in the search result.
+  # This content does not include the item's id, only URIs to the various
+  # item pages. Aggregate all the URIs and select those that match the
+  # JSON-enabled view (www.loc.gov/item/<id>)
+  #
+  # @return [Hash] the complete record
+  def get_item(doc)
+    uris = Array.new( [ doc['aka'], doc['id'], doc['url'] ] ).flatten
+    .uniq.grep(/www.loc.gov\/item/)
+    request(uris.first + '?fo=json') if !uris.empty?
```

Comment:
[Prefer `unless` to `if !`](https://github.com/bbatsov/ruby-style-guide#unless-for-negatives)